File: archive/path-crossing.js
const isPathCrossing = function(path) {
  let x = 0
  let y = 0

  let mem = {
    '[0,0]': false
  }

  let ans = false
  path.split('').forEach(d => {
    if (d === 'N') {y++}
    if (d === 'S') {y--}
    if (d === 'E') {x++}
    if (d === 'W') {x--}

    if (mem[`[${x},${y}]`] === undefined) {
      mem[`[${x},${y}]`] = false
    } else {
      ans = true
    }
  })

  return ans
}

console.log(isPathCrossing("NES"))

File: go-archive/path-crossing.go
func isPathCrossing(path string) bool {
    x:=0; y:=0; mem:=map[string]bool{"0-0": true}

    for i:=0; i < len(path); i++ {
        if path[i] == 'N' {x++}
        if path[i] == 'S' {x--}
        if path[i] == 'W' {y++}
        if path[i] == 'E' {y--}

        k:=strconv.Itoa(x) + "-" + strconv.Itoa(y)
        if _, isHas := mem[k]; isHas {
            return true
        } else {
            mem[k] = true
        }
    }

    return false
}